### ISSUE TITLE
Support month / year only selection in datepicker

### ISSUE THREAD

**Issue by mcailleaux** [angular/components#14202]
#### Bug, feature request, or proposal:
Bug

####  What is the expected behavior?
When you configure datepicker with only month and year, the click on calendar header must switch between month and years picker

####  What is the current behavior?
When you configure datepicker with only month and year, the click on calendar header switch on days picker

####  What are the steps to reproduce?
On the showroom : https://material.angular.io/components/datepicker/overview
go to the month picker, open calendar and click on year in the header

#### What is the use-case or motivation for changing an existing behavior?
To have a real month picker

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
6+, 7+

Is there anything else we should know?

**Comment by gopigopi096** [comment]
// I have solved using directive

import { Directive, HostBinding, ElementRef, HostListener, Renderer2 } from '@angular/core';

@Directive({
  selector: '[appDateDirective]'
})
export class DateDirective {

  constructor(private elementRef: ElementRef, private renderer2: Renderer2) {
    console.log(elementRef);
    console.log(elementRef.nativeElement.getAttribute('aria-label'));
  }

  @HostBinding('class.mat-calendar-period-button')
  get getPeriodButton() {
    const node = document.querySelector('*[aria-label^="Choose date"]');
    if (node) {
      this.renderer2.addClass(node, 'removePeriod');
    }
    return true;
  }

}

<mat-form-field>
  <input matInput [matDatepicker]="dp" placeholder="Month and year" [formControl]="date">
  <mat-datepicker-toggle matSuffix [for]="dp"></mat-datepicker-toggle>
  <mat-datepicker #dp id="datePick"
                  startView = "multi-year" appDateDirective
                  (yearSelected)="chosenYearHandler($event)"
                  (monthSelected)="chosenMonthHandler($event, dp)"
                  panelClass="example-month-picker">
  </mat-datepicker>
</mat-form-field>

**Comment by gargri** [comment]
if u want only month in datepicker so in html file u have to change startView="multi-year" to startView="year"
it will work correctly

**Comment by siebmanb** [comment]
@mcailleaux is your bug the fact that after choosing a year and month we shortly see a calendar for days before it fades away?

**Comment by mcailleaux** [comment]
@siebmanb no, the bug is when you are in a month picker mode, the clic on year header switch between day picker and year picker, it is more logical to switch between month and year

I make this video to illustrate the problem
[BugMonthPicker.zip](https://github.com/angular/components/files/5594965/BugMonthPicker.zip)
